src/uu/dmesg/src/dmesg.rs

```diff
@@ -0,0 +1,137 @@
+use clap::{crate_version, Arg, ArgAction, Command};
+use regex::Regex;
+use std::fs;
+use uucore::error::UResult;
+
+mod json;
+
+#[uucore::main]
+pub fn uumain(args: impl uucore::Args) -> UResult<()> {
+    let mut dmesg = Dmesg::new();
+    let matches: clap::ArgMatches = uu_app().try_get_matches_from(args)?;
+    if let Some(kmsg_file) = matches.get_one::<String>(options::KMSG_FILE) {
+        dmesg.kmsg_file = kmsg_file;
+    }
+    if matches.get_flag(options::JSON) {
+        dmesg.output_format = OutputFormat::Json;
+    }
+    dmesg.parse()?.print();
+    Ok(())
+}
+
+pub fn uu_app() -> Command {
+    Command::new(uucore::util_name())
+        .version(crate_version!())
+        .arg(
+            Arg::new(options::KMSG_FILE)
+                .short('K')
+                .long("kmsg-file")
+                .action(ArgAction::Set),
+        )
+        .arg(
+            Arg::new(options::JSON)
+                .short('J')
+                .long("json")
+                .action(ArgAction::SetTrue),
```

Comment:
Can you please add some help texts for the two options?

Reply:
I added help texts in commit 635c695.